pyiron_base/state/check.py

```diff
@@ -0,0 +1,80 @@
+import os
+import stat
+import glob
+import pandas
+
+try:
+    from conda.cli import python_api
+
+    conda_imported_successful = True
+except ImportError:
+    conda_imported_successful = False
+
+from pyiron_base import state
+
+conda_package_dict = {
+    "atomicrex": "atomicrex",
+    "cp2k": "cp2k",
+    "damask": "damask",
+    "mlip": "mlip",
+    "lammps": "lammps",
+    "quantumespresso": "qe",
+    "randspg": "randspg",
+    "runner": "runner",
+    "sphinx": "sphinxdft",
+}
+
+
+def check_for_conda_package(name):
+    lst = [l for l in python_api.run_command("list")[0].split("\n") if name + " " in l]
+    if len(lst) == 0:
+        return False
+    else:
+        return True
+
+
+def check_executable_bit(resource_paths):
+    def check_bit(script_path):
+        filemode = os.stat(script_path).st_mode
+        return bool(
+            filemode & stat.S_IXUSR
+            or filemode & stat.S_IXGRP
+            or filemode & stat.S_IXOTH
+        )
+
+    path_lst = []
+    for res_path in resource_paths:
+        path_lst += glob.glob(res_path + "/*/*/*.sh")
```

Comment:
This feels very fragile. It looks to me like this requires all executables to be at exactly the third level of depth. This may be the case right now, but so we specify or document this anywhere? Do we anticipate other sh files appearing here, or is it safe to run an endswith command recursively inside the resource path instead?

Reply:
Right now it needs to be exactly at the third level because the expected structure is `/$resources/$module_name/bin/$exe.sh`.  The documentation for this is [here](https://pyiron.readthedocs.io/en/latest/source/installation.html#use-your-own-executable-for-lammps-s-phi-nx-or-gpaw), though it could be more explicit.

I have a little class that codifies this in [tinybase](https://github.com/pyiron/pyiron_contrib/blob/tinybase/pyiron_contrib/tinybase/shell.py#L24) which could be moved to base.  Replacing this everywhere might be a bit more effort, so imo it's ok to do it in a separate PR.

Reply:
I'm ok with codefying it as long as we're sufficiently explicit. One option I see is to search for scripts at exactly three levels deep and also searching for them anywhere in the resource tree, then giving the user a warning if the lists aren't the same.

Actually, I think I really like that, and it should be straightforward to make a test setup in static to make sure both aspects work.

Similarly we should also be checking if non-supported codes have directories. I'm on mobile so I can't easily bail on this comment to check if that's already done, but this would be another nice thing to give a warning about.